packages/@ember/-internals/metal/lib/mixin.ts

```diff
@@ -575,6 +576,9 @@ export default class Mixin {
     this.ownerConstructor = undefined;
     this._without = undefined;
 
+    // This exists for objects that `Object.seal` or `Object.freeze` in DEBUG only
```

Comment:
Perhaps I could add a better comment here.  `seal` is the one that prevents adding.  `freeze` also prevents mutating, so if that was the case for the mixin, this would still fail in ember-data.

Perhaps

```
// We eagerly add INIT_FACTORY to avoid issues in DEBUG as a result of Object.seal(mixin)
```